dynawo/3rdParty/adept/patch/MSVC/adept.patch

```diff
@@ -0,0 +1,269 @@
+diff -urpN adept-2.0.5/benchmark/animate.cpp adept-2.0.5.patch/benchmark/animate.cpp
+--- adept-2.0.5/benchmark/animate.cpp	2015-07-09 23:39:41.000000000 +0200
++++ adept-2.0.5.patch/benchmark/animate.cpp	2019-10-08 12:18:24.421492600 +0200
+@@ -10,7 +10,12 @@
+ 
+ #include <string>
+ #include <iostream>
++#ifdef _MSC_VER
++#define NOMINMAX
++#include <windows.h>
++#else
+ #include <time.h>
++#endif
+ 
+ #include "advection_schemes.h"
+ 
+@@ -37,9 +42,11 @@ main(int argc, char** argv)
+   std::string line;
+   line.resize(NX);
+ 
++#ifndef _MSC_VER
+   timespec t;
+   t.tv_sec = 0;
+   t.tv_nsec = 20000000;
++#endif
+ 
+   for (int i = 0; i < NX; i++) q1[i] = (0.5+0.5*sin((i*2.0*pi)/(NX-1.5)))+0.0001;
+   for (int k = 0; k < cycles*NX/(nt*dt); k++) {
+@@ -74,7 +81,11 @@ main(int argc, char** argv)
+       std::cout << line << "\n";
+       std::cout.flush();
+     }
++#ifdef _MSC_VER
++	Sleep(20);
++#else
+     nanosleep(&t, 0);
++#endif
+     //toon(nt, dt, q1, q2);
+     lax_wendroff(nt, dt, q1, q2);
+     double* tmp = q1;
+diff -urpN adept-2.0.5/CMakeLists.txt adept-2.0.5.patch/CMakeLists.txt
+--- adept-2.0.5/CMakeLists.txt	2019-10-08 12:17:46.751573706 +0200
++++ adept-2.0.5.patch/CMakeLists.txt	2019-10-09 10:34:57.338989567 +0200
+@@ -1,7 +1,15 @@
+ cmake_minimum_required(VERSION 3.4.3 FATAL_ERROR)
+ 
++if(${CMAKE_VERSION} VERSION_GREATER "3.15.0")
++  if(POLICY CMP0091)
++    cmake_policy(SET CMP0091 NEW)
++  endif()
++endif()
++
+ project(adept)
+ 
++SET(MSVC_STATIC_RUNTIME_LIBRARY OFF CACHE BOOL "Choose either ON or OFF.")
++
+ include(CTest)
+ 
+ file(READ "include/adept/base.h" _file_content)
+@@ -12,6 +20,14 @@ message("ADEPT_VERSION_STRING=${ADEPT_VE
+ 
+ option(BUILD_SHARED_LIBS "Build shared libraries (.dll/.so) instead of static ones (.lib/.a)" ON)
+ 
++if(MSVC)
```

Comment:
This might be very hard to maintain when adept evolves...

Reply:
This is more a general concern: the number of patches increased dramatically for each 3rd party and evolving to a new version might be subtantially harder than before (and it was already a pain...)

Reply:
We knew it was the price to pay to be windows compatible... we can only try to submit PR to upstream projects to avoid this pain...